src/runtime_src/core/edge/user/aie/common_layer/adf_runtime_api.cpp

```diff
@@ -0,0 +1,654 @@
+        else //AIE RTP
+        {
+            // For connected async input RTP, only allow graph update before graph run
+            if (pRTPConfig->isAsync)
+            {
+                if (isRunning)
+                    return errorMsg(err_code::user_error, "ERROR: adf::graph::update to connected asynchronous input RTP is not allowed during graph run.");
+            }
+            // Does not support connected sync input RTP
+            else
+                return errorMsg(err_code::user_error, "ERROR: adf::graph::update to connected synchronous input RTP is not supported.");
+        }
+    }
+
+    return err_code::ok;
+}
+
+err_code graph_api::update(const rtp_config* pRTPConfig, const void* pValue, size_t numBytes)
+{
+    ///////////////////////////// Error Checking //////////////////////////////
+
+    err_code ret = checkRTPConfigForUpdate(pRTPConfig, pGraphConfig, numBytes, isRunning);
+    if (ret != err_code::ok)
+        return ret;
+
+    ///////////////////////////// Configuration //////////////////////////////
+
+    size_t numReservedRows = config_manager::s_num_reserved_rows;
+    XAie_LocType selectorTile = XAie_TileLoc(pRTPConfig->selectorColumn, pRTPConfig->selectorRow + numReservedRows + 1);
+    XAie_LocType pingTile = XAie_TileLoc(pRTPConfig->pingColumn, pRTPConfig->pingRow + numReservedRows + 1);
+    XAie_LocType pongTile = XAie_TileLoc(pRTPConfig->pongColumn, pRTPConfig->pongRow + numReservedRows + 1);
+
+    // Do NOT lock async RTP when graph is suspended; otherwise, it may deadlock. We don't support synchronous RTP in suspended mode
+    bool bAcquireLock = !(pRTPConfig->isAsync && !isRunning);
+
+    int8_t acquireVal = (pRTPConfig->isAsync ? XAIE_LOCK_WITH_NO_VALUE : ACQ_WRITE); //Versal
+    int8_t releaseVal = REL_READ; //Versal
+
+    ///////////////////////////// RTP update operation //////////////////////////////
+
+    infoMsg("Updating RTP value to port " + pRTPConfig->portName);
+
+    int driverStatus = AieRC::XAIE_OK; //0
+
+    // sync ports acquire selector lock for WRITE, async ports acquire selector lock unconditionally
+    if (pRTPConfig->hasLock && bAcquireLock)
+        driverStatus |= XAie_LockAcquire(config_manager::s_pDevInst, selectorTile, XAie_LockInit(pRTPConfig->selectorLockId, acquireVal), LOCK_TIMEOUT);
+
+    // Read the selector value
+    u32 selector;
+    driverStatus |= XAie_DataMemRdWord(config_manager::s_pDevInst, selectorTile, pRTPConfig->selectorAddr, ((u32*)&selector));
+    selector = 1 - selector;
+
+    if (selector == 1) //pong
+    {
+        // sync ports acquire buffer lock for WRITE, async ports acquire buffer lock unconditionally
+        if (pRTPConfig->hasLock && bAcquireLock)
+            driverStatus |= XAie_LockAcquire(config_manager::s_pDevInst, pongTile, XAie_LockInit(pRTPConfig->pongLockId, acquireVal), LOCK_TIMEOUT);
+
+        driverStatus |= XAie_DataMemBlockWrite(config_manager::s_pDevInst, pongTile, pRTPConfig->pongAddr, pValue, numBytes);
```

Comment:
/proj/xsjhdstaff5/yliu/project/featrom/XRT/src/runtime_src/core/edge/user/aie/common_layer/adf_runtime_api.cpp:396:108: error: invalid conversion from 'const void*' to 'void*' [-fpermissive]
|   396 |         driverStatus |= XAie_DataMemBlockWrite(config_manager::s_pDevInst, pongTile, pRTPConfig->pongAddr, pValue, numBytes);
|       |                                                                                                            ^~~~~~
|       |                                                                                                            |
|       |                                                                                                            const void* 

```suggestion
        driverStatus |= XAie_DataMemBlockWrite(config_manager::s_pDevInst,      pongTile, const_cast<char *>(pRTPConfig->pongAddr), pValue, numBytes);
```

Reply:
done